### PR TITLE
Migrated all table and column names to lower snake case

### PR DESCRIPTION
**Pull request by joostlek**
## Summary

I want to improve the consistency in database names, in this merge I will cover the table names.
## Link to pull request in Documentation repository

Documentation: home-assistant/companion.home-assistant#

## Any other notes

I probably have to do another one for the column names since it did not want to build 2 migrations at the same time.

### CODE REVIEW

**Comment by jpelgrom** [comment]
> I probably have to do another one for the column names since it did not want to build 2 migrations at the same time.

Clearly you're still working on it but as you're using 2 auto migrations right now for table names and column names, why not do it in 1 version/migration?

**Comment by joostlek** [comment]
> > I probably have to do another one for the column names since it did not want to build 2 migrations at the same time.
> 
> Clearly you're still working on it but as you're using 2 auto migrations right now for table names and column names, why not do it in 1 version/migration?

For some reason it gave a lot of exceptions locally, probably had to do with the order of the migrations (since you need to reference the table name in the rename column annotation), but it seems to be working now

File: common/src/main/java/io/homeassistant/companion/android/database/AppDatabase.kt [hunk 1]
```diff
@@ -527,6 +533,99 @@ abstract class AppDatabase : RoomDatabase() {
...
+            RenameColumn(
+                tableName = "Authentication_List",
+                fromColumnName = "Username",
+                toColumnName = "username"
+            ),
+            RenameColumn(
+                tableName = "Authentication_List",
+                fromColumnName = "Password",
+                toColumnName = "password"
+            ),
```
Comment (jpelgrom) [inline common/src/main/java/io/homeassistant/companion/android/database/AppDatabase.kt:546]: SQLite is case insensitive, but if you're going to rename columns here please also update the relevant data object.

**Comment by jpelgrom** [comment]
Did you actually test the migration? It is crashing on launch on upgrade because the migration you've provided doesn't work (it is missing columns).

<details>
<summary>Example</summary>

Crash stacktrace:

```
2023-01-02 17:50:55.491 5125-5125/? E/AndroidRuntime: FATAL EXCEPTION: main
    Process: io.homeassistant.companion.android.debug, PID: 5125
    java.lang.RuntimeException: Unable to create application io.homeassistant.companion.android.HomeAssistantApplication: android.database.sqlite.SQLiteConstraintException: NOT NULL constraint failed: _new_settings.sensor_update_frequency (code 1299 SQLITE_CONSTRAINT_NOTNULL)
        at android.app.ActivityThread.handleBindApplication(ActivityThread.java:6764)
        at android.app.ActivityThread.-$$Nest$mhandleBindApplication(Unknown Source:0)
        at android.app.ActivityThread$H.handleMessage(ActivityThread.java:2133)
        at android.os.Handler.dispatchMessage(Handler.java:106)
        at android.os.Looper.loopOnce(Looper.java:201)
        at android.os.Looper.loop(Looper.java:288)
        at android.app.ActivityThread.main(ActivityThread.java:7872)
        at java.lang.reflect.Method.invoke(Native Method)
        at com.android.internal.os.RuntimeInit$MethodAndArgsCaller.run(RuntimeInit.java:548)
        at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:936)
     Caused by: android.database.sqlite.SQLiteConstraintException: NOT NULL constraint failed: _new_settings.sensor_update_frequency (code 1299 SQLITE_CONSTRAINT_NOTNULL)
        at android.database.sqlite.SQLiteConnection.nativeExecuteForChangedRowCount(Native Method)
        at android.database.sqlite.SQLiteConnection.executeForChangedRowCount(SQLiteConnection.java:913)
        at android.database.sqlite.SQLiteSession.executeForChangedRowCount(SQLiteSession.java:756)
        at android.database.sqlite.SQLiteStatement.executeUpdateDelete(SQLiteStatement.java:67)
        at android.database.sqlite.SQLiteDatabase.executeSql(SQLiteDatabase.java:2087)
        at android.database.sqlite.SQLiteDatabase.execSQL(SQLiteDatabase.java:2008)
        at androidx.sqlite.db.framework.FrameworkSQLiteDatabase.execSQL(FrameworkSQLiteDatabase.java:265)
        at io.homeassistant.companion.android.database.AppDatabase_AutoMigration_36_37_Impl.migrate(AppDatabase_AutoMigration_36_37_Impl.java:27)
        at androidx.room.RoomOpenHelper.onUpgrade(RoomOpenHelper.java:99)
        at androidx.sqlite.db.framework.FrameworkSQLiteOpenHelper$OpenHelper.onUpgrade(FrameworkSQLiteOpenHelper.java:183)
        at android.database.sqlite.SQLiteOpenHelper.getDatabaseLocked(SQLiteOpenHelper.java:416)
        at android.database.sqlite.SQLiteOpenHelper.getWritableDatabase(SQLiteOpenHelper.java:316)
        at androidx.sqlite.db.framework.FrameworkSQLiteOpenHelper$OpenHelper.getWritableSupportDatabase(FrameworkSQLiteOpenHelper.java:151)
        at androidx.sqlite.db.framework.FrameworkSQLiteOpenHelper.getWritableDatabase(FrameworkSQLiteOpenHelper.java:112)
        at androidx.room.RoomDatabase.inTransaction(RoomDatabase.java:706)
        at androidx.room.RoomDatabase.assertNotSuspendingTransaction(RoomDatabase.java:483)
        at io.homeassistant.companion.android.database.sensor.SensorDao_Impl.getSettings(SensorDao_Impl.java:1224)
        at io.homeassistant.companion.android.HomeAssistantApplication.onCreate(HomeAssistantApplication.kt:181)
        at android.app.Instrumentation.callApplicationOnCreate(Instrumentation.java:1277)
        at android.app.ActivityThread.handleBindApplication(ActivityThread.java:6759)
        at android.app.ActivityThread.-$$Nest$mhandleBindApplication(Unknown Source:0) 
        at android.app.ActivityThread$H.handleMessage(ActivityThread.java:2133) 
        at android.os.Handler.dispatchMessage(Handler.java:106) 
        at android.os.Looper.loopOnce(Looper.java:201) 
        at android.os.Looper.loop(Looper.java:288) 
        at android.app.ActivityThread.main(ActivityThread.java:7872) 
        at java.lang.reflect.Method.invoke(Native Method) 
        at com.android.internal.os.RuntimeInit$MethodAndArgsCaller.run(RuntimeInit.java:548) 
        at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:936) 
```

This is failing because the generated migration doesn't update all fields correctly:

```kotlin
database.execSQL("CREATE TABLE IF NOT EXISTS `_new_settings` (`id` INTEGER NOT NULL, `websocket_setting` TEXT NOT NULL, `sensor_update_frequency` TEXT NOT NULL, PRIMARY KEY(`id`))");
database.execSQL("INSERT INTO `_new_settings` (`id`,`websocket_setting`) SELECT `id`,`websocketSetting` FROM `settings`");
database.execSQL("DROP TABLE `settings`");
```
</details>

**Comment by joostlek** [comment]
For me on my virtual device it just boots and works but after your comment I am not sure if the migrations actually have taken place, is there any way to check or force the migrations to happen?

**Comment by jpelgrom** [comment]
Room always ensures the app database is using the latest version, but that doesn't mean it has run the migration(s). The easiest option is probably to install the app from `master` and launch it first, then switch to your branch and install and launch it. If you do a clean install from your branch there is nothing to migrate so it isn't run ;)

**Comment by joostlek** [comment]
> Room always ensures the app database is using the latest version, but that doesn't mean it has run the migration(s). The easiest option is probably to install the app from `master` and launch it first, then switch to your branch and install and launch it. If you do a clean install from your branch there is nothing to migrate so it isn't run ;)

So I tried this and the first time it complained about a table being there, so I did the clean install of master. Then I switched to my branch and click relaunch and it worked normally. Am I doing something wrong?

**Comment by jpelgrom** [comment]
> the first time it complained about a table being there

Room will complain if you try to downgrade as it is not a supported scenario :) Otherwise what you're describing sounds correct, yet it is still generating invalid migration code as far as I can tell and crashing on launch. I did change the sensor update frequency setting after doing a clean install from `master`, to test that something is written to the database and correctly migrates.

Looking at the generated migration not working with 2 renames, I believe this PR is hitting [a bug resolved by this commit](https://android-review.googlesource.com/c/platform/frameworks/support/+/2215956), which is included in the upcoming 2.5.0 release of Room. Possible fixes would be
1. wait a while until the next version is released,
2. switch to the 2.5.0-rc01 version, or
3. write a manual migration.

Option 1 or 2 would be preferred in my opinion.

**Comment by jpelgrom** [comment]
Room was updated to 2.5.0 in #3216, if you rebase your branch on master or merge the latest changes the code should be good.

**Review by jpelgrom — APPROVED** [review]
Did a local rebase on the latest master (96cf4f8) to test the changes; migration is now working correctly and no change in generated json file, so this should be good to merge.

**Comment by joostlek** [comment]
Thank you :)

**Review by JBassett — APPROVED** [review]

**Pull request merged**